tools/linter/adapters/import_linter.py

```diff
@@ -0,0 +1,391 @@
+            "nis",
+            "nntplib",
+            "ntpath",
+            "nturl2path",
+            "numbers",
+            "opcode",
+            "operator",
+            "optparse",
+            "os",
+            "ossaudiodev",
+            "parser",
+            "pathlib",
+            "pdb",
+            "pickle",
+            "pickletools",
+            "pipes",
+            "pkgutil",
+            "platform",
+            "plistlib",
+            "poplib",
+            "posix",
+            "posixpath",
+            "pprint",
+            "profile",
+            "pstats",
+            "pty",
+            "pwd",
+            "py_compile",
+            "pyclbr",
+            "pydoc",
+            "pydoc_data",
+            "queue",
+            "quopri",
+            "random",
+            "re",
+            "readline",
+            "reprlib",
+            "resource",
+            "rlcompleter",
+            "runpy",
+            "sched",
+            "secrets",
+            "select",
+            "selectors",
+            "shelve",
+            "shlex",
+            "shutil",
+            "signal",
+            "site",
+            "smtpd",
+            "smtplib",
+            "sndhdr",
+            "socket",
+            "socketserver",
+            "spwd",
+            "sqlite3",
+            "sre_compile",
+            "sre_constants",
+            "sre_parse",
+            "ssl",
+            "stat",
+            "statistics",
+            "string",
+            "stringprep",
+            "struct",
+            "subprocess",
+            "sunau",
+            "symbol",
+            "symtable",
+            "sys",
+            "sysconfig",
+            "syslog",
+            "tabnanny",
+            "tarfile",
+            "telnetlib",
+            "tempfile",
+            "termios",
+            "test",
+            "textwrap",
+            "this",
+            "threading",
+            "time",
+            "timeit",
+            "tkinter",
+            "token",
+            "tokenize",
+            "trace",
+            "traceback",
+            "tracemalloc",
+            "tty",
+            "turtle",
+            "turtledemo",
+            "types",
+            "typing",
+            "unicodedata",
+            "unittest",
+            "urllib",
+            "uu",
+            "uuid",
+            "venv",
+            "warnings",
+            "wave",
+            "weakref",
+            "webbrowser",
+            "winreg",
+            "winsound",
+            "wsgiref",
+            "xdrlib",
+            "xml",
+            "xmlrpc",
+            "xxsubtype",
+            "zipapp",
+            "zipfile",
+            "zipimport",
+            "zlib",
+            "zoneinfo",
+        ]
+    )
+
+# Add the allowed third party libraries.
+_MODULE_NAME_ALLOW_LIST.update(
+    [
+        "torch",
+        "sympy",
+        "torch_xla",
+        "_pytest",
+        "functorch",
+        "einops",
+        "the",
```

Comment:
I don't see any "import the" statements in torch/_dynamo -- how did this come up?

Reply:
Oh lol is this literally "from the" in comments?

Reply:
LOL yep: https://github.com/pytorch/pytorch/blob/fda43c98d1bc82eeb493f6e79a1fd5f1636474a5/torch/_dynamo/resume_execution.py#L465

Ughhh I probably have to replace line-by-line regex with some AST-aware thing, but thankfully the linter does seem to provide parsing and processing for that, lemme look more.

Reply:
Maybe just a "from foo import y" regex

Reply:
That'll still get picked up in a multi-line string... But maybe suffices in practice, I'll make some decision based on complexity of the alternative.

Reply:
Ended up using `_linter.PythonFile` and some basic token checks, quite simple.

Reply:
Nice!